src/uct/ib/mlx5/ib_mlx5.inl

```diff
@@ -72,28 +84,155 @@ uct_ib_mlx5_update_db_cq_ci(uct_ib_mlx5_cq_t *cq)
 }
 
 
+/**
+ * Fill the given CQE by unique data from given mini_cqe.
+ *
+ * @param cqe       The destination CQE.
+ * @param mini_cqe  The source mini CQE.
+ * @param cq_decomp The structure that contains info about CQE compression block.
+ */
+static UCS_F_ALWAYS_INLINE void
+uct_ib_mlx5_iface_cqe_decompress_fill_unique_fields(
+    struct mlx5_cqe64 *cqe, uct_ib_mlx5_mini_cqe8_t *mini_cqe,
+    uct_ib_mlx5_cq_decompress_t *cq_decompress)
+{
+    uint32_t host_sop_drop_qpn;
+    uint8_t  host_s_wqe_opcode;
+
+    /* TODO: Think about the checksum field decompression.
+     * The checksum field does not copied from src dst
+     * because of absence of the mlx5_cqe64::check_sum field on some systems.
+     */
+    cqe->byte_cnt           = mini_cqe->byte_cnt;
+    ucs_assert(!(cqe->op_own & UCT_IB_MLX5_CQE_OP_OWN_ERR_MASK));
+    if ((cqe->op_own >> 4) == MLX5_CQE_REQ) { /* Requester */
+        cqe->wqe_counter    = mini_cqe->wqe_counter;
+
+        /* Set s_wqe_opcode */
+        host_sop_drop_qpn   = ntohl(cqe->sop_drop_qpn);
+        host_s_wqe_opcode   = ntohs(mini_cqe->s_wqe_opcode);
+        host_sop_drop_qpn  &= UCS_MASK(UCT_IB_QPN_ORDER);
+        host_sop_drop_qpn  |= host_s_wqe_opcode << UCT_IB_QPN_ORDER;
+        cqe->sop_drop_qpn   = htonl(host_sop_drop_qpn);
```

Comment:
can we perform all these manipulations in network endianess, w/o converting to host endianess?
also opcode is 1byte so does not need conversion

Reply:
Right, need to do `uint32_t net_sop = htonl(s_wqe_opcode << UCT_IB_QPN_ORDER)`
see https://godbolt.org/z/bc17f4qT9